h2o-py/h2o/backend/connection.py

```diff
@@ -125,14 +240,16 @@ def open(server=None, url=None, ip=None, port=None, https=None, auth=None, verif
             ip = server.ip
             port = server.port
             scheme = server.scheme
+            context_path = ''
         elif url is not None:
             assert_is_type(url, str)
             assert_is_type(ip, None, "`ip` should be None when `url` parameter is supplied")
             # We don't allow any Unicode characters in the URL. Maybe some day we will...
-            match = assert_matches(url, r"^(https?)://((?:[\w-]+\.)*[\w-]+):(\d+)/?$")
+            match = assert_matches(url, r"^(https?)://((?:[\w-]+\.)*[\w-]+):(\d+)/?([\w-]*)?$")
```

Comment:
do we want to allow context path of several levels of depth? `"http://localhost:80/my/h2o/server"`